Week1/databases/homework/index.js

```diff
@@ -0,0 +1,182 @@
+    }
+  }
+
+  async function fillTables() {
+    try {
+      const data = await readFile('data.json', 'utf8');
+      const { cities, countries } = JSON.parse(data);
+
+      const cityCount = await executeQuery(`select count(id) as numberOfCities from cities`);
+      const isCitiesTableEmpty = cityCount[0].numberOfCities === 0 ? true : false;
+
+      if (isCitiesTableEmpty) {
+        cities.forEach(city => {
+          connection.query(
+            `insert into cities set 
+              Name = "${city.name}", 
+              CountryCode = "${city.country_code}", 
+              District = "${city.district}", 
+              Population = ${city.population}`,
+            error => {
+              if (error) {
+                console.error(
+                  `There was an error while inserting into cities table: ${error.code} - ${error.message}`,
+                );
+              }
+            },
+          );
+        });
+      }
+
+      const countryCount = await executeQuery(
+        `select count(id) as numberOfCountries from countries`,
+      );
+      const isCountriesTableEmpty = countryCount[0].numberOfCountries === 0 ? true : false;
+
+      if (isCountriesTableEmpty) {
+        countries.forEach(country => {
+          connection.query(
+            `insert into countries set 
+              Name = "${country.name}", 
+              Continent = "${country.continent}",
+              Region = "${country.region}", 
+              SurfaceArea = ${country.surface_area}, 
+              IndepYear = ${country.indep_year},
+              Population = ${country.population},
+              LifeExpectancy = ${country.life_expectancy},
+              GNP = ${country.gnp},
+              GNPOld = ${country.gnp_old},
+              LocalName = "${country.local_name}",
+              GovernmentForm = "${country.government_form}",
+              HeadOfState = "${country.head_of_state}",
+              Capital = ${country.capital}`,
+
+            error => {
+              if (error) {
+                console.error(
+                  `There was an error while inserting into countries table: ${error.code} - ${error.message}`,
+                );
+              }
+            },
+          );
+        });
+      }
+    } catch (error) {
+      console.error(`There was an error: ${error.code} - ${error.message}`);
+    }
+  }
+
+  async function makeQuery(query) {
+    try {
+      const result = await executeQuery(query);
+      console.log(JSON.stringify(result, null, 2));
+    } catch (error) {
+      console.error(
+        `There was an error while making the query: ${query} \n The error is: ${error}`,
+      );
+    }
+  }
+
+  async function makeQueries() {
+    await initDatabase();
+    await fillTables();
+
+    // QUERIES
+
+    console.log(`\n The names of the countries with population greater than 8 million \n`);
+    await makeQuery(`select name from countries where population > 8000000`);
```

Comment:
Great job! All of the queries are good :)

Reply:
Thanks!